openedx/core/djangoapps/content_tagging/types.py

```diff
@@ -1,8 +1,26 @@
 """
 Types used by content tagging API and implementation
 """
-from typing import Union
+from __future__ import annotations
+from typing import Union, Dict, List
 
+from attrs import define
 from opaque_keys.edx.keys import LearningContextKey, UsageKey
+from openedx_tagging.core.tagging.models import ObjectTag, Taxonomy
+from xblock.core import XBlock
 
 ContentKey = Union[LearningContextKey, UsageKey]
+
+ObjectTagByTaxonomyIdDict = Dict[int, List[ObjectTag]]
+ObjectTagByObjectIdDict = Dict[str, ObjectTagByTaxonomyIdDict]
+TaxonomyDict = Dict[int, Taxonomy]
+
+
+@define
+class TaggedContent:
+    """
+    A tagged content, with its tags and children.
+    """
+    xblock: XBlock  # ToDo: Check correct type here
```

Comment:
Storing the whole `XBlock` instance for the whole course tree could use a lot of memory. You only really need to store the `usage_id` and the `display_name`, so I would suggest just storing those instead of `xblock`